### PR TITLE
Add Boolean field in star models

### PR DESCRIPTION
**Pull request by RishabhJain2018**
@deshraj Please review the PR.

### CODE REVIEW

File: apps/challenges/models.py [hunk 1]
```diff
@@ -228,6 +228,7 @@ class StarChallenge(TimeStampedModel):
     """
     user = models.ForeignKey(User)
     challenge = models.ForeignKey(Challenge)
+    is_starred = models.BooleanField(default=False)
```
Comment (taranjeet) [inline apps/challenges/models.py]: Entry in this model will be made only when a user stars a particular challenge. Why an extra boolean field here ?

Comment (RishabhJain2018) [inline apps/challenges/models.py]: Without this field when a user will unstar a challenge then we have to delete the entry from this model but now with this boolean field when a user will unstar it then we will toggle this field which will maintain our database schema indexes. Moreover, we can also get the analytics like how many people starred the challenge and how many are currently active.

Comment (taranjeet) [inline apps/challenges/models.py]: I thought we were not tracking unstar action.

Comment (deshraj) [inline apps/challenges/models.py]: We have done this so that we can get information in future if we need some analytics for this. :)

**Pull request merged**